### PR TITLE
LPS-180189 NoSuchUserException when adding PLO with liferay-online.properties configured after migrate from u62 to u69

### PR DESCRIPTION
**Pull request by jgarcialr**

### CODE REVIEW

File: modules/apps/portal-language/portal-language-override-service/src/main/java/com/liferay/portal/language/override/service/impl/PLOEntryLocalServiceImpl.java [hunk 1]
```diff
@@ -61,9 +60,7 @@ public PLOEntry addOrUpdatePLOEntry(
 
 			ploEntry.setCompanyId(companyId);
 
-			User user = _userLocalService.getUser(userId);
-
-			ploEntry.setUserId(user.getUserId());
+			ploEntry.setUserId(userId);
```
Comment (drewbrokke) [inline modules/apps/portal-language/portal-language-override-service/src/main/java/com/liferay/portal/language/override/service/impl/PLOEntryLocalServiceImpl.java:63]: The call to `get` is intentional. What is the reason that an invalid `userId` is being passed in here? 

If we allow setting to an invalid `userId`, then we might just see this exception show up in a random spot elsewhere when they read this property expecting it to be valid.

**Review by drewbrokke — CHANGES_REQUESTED** [review]
@jgarcialr thank you for sending this change! I do have a concern though, please see my inline comment.

File: modules/apps/portal-language/portal-language-override-service/src/main/java/com/liferay/portal/language/override/service/impl/PLOEntryLocalServiceImpl.java (hunk 1 above)
Comment (jgarcialr) [reply modules/apps/portal-language/portal-language-override-service/src/main/java/com/liferay/portal/language/override/service/impl/PLOEntryLocalServiceImpl.java:63]: Hi @drewbrokke ,

If the user is from a secondary company, when registering the objectdefinitions, there's a call made from objectDeployerImpl to addorUpdate a PLO.

As those users are not loaded yet, this request throws the NoSuchUserException despite the user existing in the database.

**Comment by jgarcialr** [comment]
Hi @drewbrokke,

I think I can set the current companyId in threadlocal in objectdefinitiondeployer and it should work without this change.
Let me test it and I'll resend the PR to objects team.

**Comment by jgarcialr** [comment]
ci:stop

**Comment by drewbrokke** [comment]
@jgarcialr perhaps it would work to wrap these two calls? 
[ObjectDefinitionLocalServiceImpl.java#L476](https://github.com/drewbrokke/liferay-portal/blob/master/modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectDefinitionLocalServiceImpl.java?plain=1#L476)
[ObjectDefinitionLocalServiceImpl.java#L839](https://github.com/drewbrokke/liferay-portal/blob/master/modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectDefinitionLocalServiceImpl.java?plain=1#L839)

**Comment by jgarcialr** [comment]
I did the change inside deploy method, I don't know if it's better to set it in there or at the point you advise to.

**Comment by drewbrokke** [comment]
Either could work, but if it's possible that you'll run into this issue with some other service, then wrapping the calls to `deploy` would ensure that all implementations will happen inside the proper CompanyThreadLocal context.

Just my 2c

**Pull request closed without merging**